### ISSUE TITLE
Unexpected white characters when using `--funcattributes same-line`

### ISSUE THREAD

**Issue by marciniwanicki** [nicklockwood/SwiftFormat#1416]
Noticed an issue in the latest version (0.51.4), also reproducible in 0.49.9.

Config:
```
--swiftversion 5.7
--rules wrapAttributes
--funcattributes same-line
```

File before formatting:

```swift
class MyClass: NSObject {
    @objc
    func myFunction() {
        print("Testing")
    }
}
```

After formatting:
```swift
class MyClass: NSObject {
    @objc     func myFunction() {
        print("Testing")
    }
}
```

Expected output (single space between `@objc` and `func`):

```swift
class MyClass: NSObject {
    @objc func myFunction() {
        print("Testing")
    }
}
```

**Comment by nicklockwood** [comment]
@marciniwanicki fixed in 0.51.5

**Comment by marciniwanicki** [comment]
Super, thanks @nicklockwood 🙌
